src/snapshot/component_checksum_hash.rs

```diff
@@ -0,0 +1,68 @@
+use std::{
+    collections::hash_map::DefaultHasher,
+    hash::{Hash, Hasher},
+    marker::PhantomData,
+};
+
+use bevy::prelude::*;
+
+use crate::{ChecksumFlag, ChecksumPart, Rollback, SaveWorld, SaveWorldSet};
+
+/// A [`Plugin`] which will track the [`Component`] `C` on [`Rollback Entities`](`Rollback`) and ensure a
+/// [`ChecksumPart`] is available and updated. This can be used to generate a [`Checksum`](`crate::Checksum`).
+pub struct GgrsComponentChecksumHashPlugin<C>
+where
+    C: Component + Hash,
+{
+    _phantom: PhantomData<C>,
+}
+
+impl<C> Default for GgrsComponentChecksumHashPlugin<C>
+where
+    C: Component + Hash,
+{
+    fn default() -> Self {
+        Self {
+            _phantom: default(),
+        }
+    }
+}
+
+impl<C> GgrsComponentChecksumHashPlugin<C>
+where
+    C: Component + Hash,
+{
+    /// A [`System`] responsible for managing a [`ChecksumPart`] for the [`Component`] type `C`.
+    pub fn update(
+        mut commands: Commands,
+        components: Query<(&Rollback, &C), (With<Rollback>, Without<ChecksumFlag<C>>)>,
+        mut checksum: Query<&mut ChecksumPart, (Without<Rollback>, With<ChecksumFlag<C>>)>,
+    ) {
+        let mut hasher = DefaultHasher::new();
```

Comment:
> ```
> /// The default [`Hasher`] used by [`RandomState`].
> ///
> /// The internal algorithm is not specified, and so it and its hashes should
> /// not be relied upon over releases.
```

Maybe it would be better/more appropriate to use `bevy_utils::FixedState.build_hasher`, which is what reflect_hash uses?

Reply:
Didn't even know about that! Good idea!